packages/streams/logging.js

```diff
@@ -28,6 +28,51 @@ module.exports = {
   listLogFiles
 }
 
+class FileTimestampStreamWithDelete extends FileTimestampStream {
+  constructor(app, fullLogDir, filesToKeep, options){
+    super(options)
+    this.app = app
+    this.filesToKeep = filesToKeep
+    this.fullLogDir = fullLogDir
+    this.prevFilename = undefined
+  }
+
+  // This method of base class is called when new file name is contemplated
+  // So let's override it to check how many files are there and delete the oldest ones
+  newFilename() {
+    if (this.prevFilename !== this.currentFilename){ // Only do that after new file created
+      this.prevFilename = this.currentFilename
+      this.deleteOldFiles()
+    }
+    return super.newFilename()
+  }
+
+  deleteOldFiles(){
+    debug(`Checking for old log files`)
+    listLogFiles(this.app, (err, files) => {
+      if (err) {
+        console.error(err);
+      }else{
+        if (files.length > this.filesToKeep) {
+          const sortedFiles = files.sort();
+          const numToDelete =  files.length - this.filesToKeep;
+          debug(`Will delete ${numToDelete} files`)
+          for(let i = 0; i < numToDelete; i++){
+            const fileName = path.join(this.fullLogDir, sortedFiles[i])
+            debug(`Deleting ${fileName}`)
+            fs.unlink(fileName, (err) => {
+              if (err) console.error(err);
+              else debug(`${fileName} was deleted`);
+            });
+          }
+        }else{
+          debug(`Only ${files.length} out of ${this.filesToKeep} allowed. No deletion.`)
```

Comment:
For some reason I find the wording of this message hard to understand.

What if we had just one debug statement: 
```
debug(`Found ${files.length}/${this.filesToKeep} log files, deleting ${Math.max(numToDelete, 0)}`)
```

Reply:
Done